src/Miunie.Storage/JsonDataStorage.cs

```diff
@@ -76,39 +70,24 @@ public void WipeData()
             Directory.Delete(_resourcesFolder);
         }
 
-        public bool KeyExists(string collection, string key)
-        {
-            var file = GetFileNameByKey(key);
-            return LocalFileExists(String.Concat(collection, "/", file));
-        }
+        private string KeyToFullFileName(string key) => String.Format(FileTemplate, key);
 
-        private bool LocalFileExists(string file)
-        {
-            string filePath = String.Concat(_resourcesFolder, "/", file);
-            return File.Exists(filePath);
-        }
+        private string GetFullFilePath(string collection, string key) =>
+            Path.Combine(_resourcesFolder, collection, KeyToFullFileName(key));
 
-        private string GetFileNameByKey(string key)
-            => String.Format(FileTemplate, key);
+        private string GetCollectionDirectory(string collection) => Path.Combine(_resourcesFolder, collection);
 
-        private T RestoreByPath<T>(string filePath)
+        private void EnsureCollectionDirectoryExists(string collection)
         {
-            string json = GetOrCreateFileContent(filePath);
-            return JsonConvert.DeserializeObject<T>(json);
+            var collectionDir = Path.Combine(_resourcesFolder, collection);
+            Directory.CreateDirectory(collectionDir);
         }
 
-        private string GetOrCreateFileContent(string path)
+        private string GetOrCreateFileContent(string filePath)
         {
-            var filePath = String.Concat(_resourcesFolder, "/", path);
             if (File.Exists(filePath)) return File.ReadAllText(filePath);

```

Comment:
Alright, I have to admit it, I've broke the _No body should be ommited_ style rule here.
I'd appreciate if you could change this too.
Just for reference:
```cs
//Wrong
if (foo is null) return;
//Correct
if (foo is null) { return; }
//Correct
if (bar.Contains(foo))
{
    return bar;
}
```